ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/target/P2TargetUtils.java

```diff
@@ -1046,7 +1079,8 @@ private static void addReferences(IMetadataRepository repository, List<IMetadata
 		Collection<IRepositoryReference> references = repository.getReferences();
 		SubMonitor subMonitor = SubMonitor.convert(monitor, references.size() * 2);
 		for (IRepositoryReference reference : references) {
-			if (reference.getType() == IRepository.TYPE_METADATA && seen.add(reference)) {
+			if (reference.getType() == IRepository.TYPE_METADATA && reference.getOptions() == IRepository.ENABLED
```

Comment:
Good catch! According to the API docs `IRepositoryReference#getOptions` is a bitfield (even though it only supports two values at the moment).

That said, the (internal) usage of bitfields is quite cumbersome maybe we should add a (default implemented) method `isEnabled()` to the interface that query the bitfield and then can be used in a more semantic way.